src/app/dialogs/import-experiment/import-experiment.component.ts

```diff
@@ -5,126 +5,115 @@ import { filter, startWith, switchMap, take } from 'rxjs/operators';
+                    const importId: number = experimentUpload.importId;
+                    this.polling = "PENDING";
+                    this.uploading = false;
+                    interval(1000)
+                        .pipe(
+                            startWith(0),
+                            switchMap(() => this.backend.importExperimentPoll(importId)),
+                            filter(resp => resp.status != "PENDING"),
+                            take(1)
+                        )
+                        .subscribe(resp => {
+                            if (resp.status == "SUCCESS") {
+                                // close dialog after wait
+                                this.polling = "DONE"
+                                this.pollingResult = "Import successful. Forwarding to experiment...";
+                                this.waiting = true;
+                                setTimeout(() => this.dialogRef.close({ experimentId: resp.experimentId }), 2000);
+                            } else {
+                                if (resp.status == "FAILURE") {
+                                    console.error("Something went wrong. Check errors at backend.")
+                                } else {
+                                    console.error("Something went wrong. Backend returned wrong import status. Please check errors at backend.");
+                                }
+                                this.polling = "DONE";
+                                this.error = "Something went wrong.";
+                                this.reset();
+                            }
+                        })
                 } else {
-                  console.error("Something went wrong. Backend returned wrong import status. Please check errors at backend.");
+                    console.error("Something went wrong. Could not retrieve importId. Please check errors at backend.");
+                    this.error = "Something went wrong.";
+                    this.reset();
                 }
-                this.polling = "DONE";
-                this.error = "Something went wrong.";
-                this.reset();
-              }
             })
-        } else {
-          console.error("Something went wrong. Could not retrieve importId. Please check errors at backend.");
-          this.error = "Something went wrong.";
-          this.reset();
-        }
-      })
-  }
+    }
 
-  reset() {
-    this.waiting = true;
-    setTimeout(() => {
-      this.waiting = false;
-      this.error = undefined;
-      this.progress = 0;
-      this.uploadStatus = "NOT_STARTED";
-      this.polling = "NOT_STARTED";
-    }, 3000);
-  }
+    reset() {
+        this.waiting = true;
+        setTimeout(() => {
+            this.waiting = false;
+            this.error = undefined;
+            this.progress = 0;
+            this.uploadStatus = "NOT_STARTED";
+            this.polling = "NOT_STARTED";
+        }, 4000);
```

Comment:
What is this tmeout hack doing here and why was there a need to increase the delay? This does not seem like it will be robust in the future.

Reply:
The idea was to give the user time to see the error message and then automatically reset the inputs after 4 sec. Now changed to simply enable the choose file button again in case an error happens (instead of auto-reset).

Reply:
It would have been ok, if that intention was documented in code. A small comment explaining the delay was missing.
But your solution should be even better as 4 seconds to read and understand an error message can be too fast and it is hard to find a good timing for such things.